### PR TITLE
New roles decorator with matching mode integrated

### PR DESCRIPTION
**Pull request by jalorenz**
Pull Request for #45

### CODE REVIEW

File: src/guards/role.guard.ts [hunk 1]
```diff
@@ -63,8 +56,8 @@ export class RoleGuard implements CanActivate {
     // Grab access token from grant
     const accessToken: KeycloakConnect.Token = grant.access_token as any;
 
-    return allowAnyRole
-      ? roles.some(r => accessToken.hasRole(r))
-      : roles.every(r => accessToken.hasRole(r));
+    return rolesMetaData.RoleMatchingMode === RoleMatchingMode.any
```
Comment (ferrerojosh) [inline src/guards/role.guard.ts:64]: Should we add any as default or no ? Feel free to suggest or we can leave it as is.

File: src/guards/role.guard.ts [hunk 2]
```diff
@@ -27,21 +25,16 @@ export class RoleGuard implements CanActivate {
...
-    const allowAnyRole = this.reflector.get<boolean>(
-      META_ALLOW_ANY_ROLE,
-      context.getHandler(),
-    );
 
-    // No roles given, since we are permissive, allow
-    if (!roles) {
-      return true;
+    if (!rolesMetaData || rolesMetaData.roles.length == 0) {
+      return false;
```
[the remark this answers is not in the document]
Comment (jalorenz) [reply src/guards/role.guard.ts:39]: True, will change this :)

File: src/guards/role.guard.ts (hunk 1 above)
Comment (jalorenz) [reply src/guards/role.guard.ts:64]: Good point, what do you think about making `RoleMatchingMode.any` to default?

Comment (ferrerojosh) [reply src/guards/role.guard.ts:64]: Sure, but it will break behavior before this, going to add a warning when I release this.

**Review by ferrerojosh — APPROVED** [review]

**Pull request merged**